lib/generators/shopify_app/hmac_controller/templates/hmac_verification_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+class HmacVerificationController < ApplicationController
+  protect_from_forgery with: :null_session
+
+  before_action :verify_request
+
+  private
+
+  def verify_request
+    hmac_header = request.headers['HTTP_X_SHOPIFY_HMAC_SHA256']
+    request_body = request.body.read
+    secret = ShopifyApp.configuration.secret
+    digest = OpenSSL::Digest.new('sha256')
+
+    calculated_hmac = Base64.strict_encode64(OpenSSL::HMAC.digest(digest, secret, request_body))
+    head(:unauthorized) unless calculated_hmac == hmac_header
```

Comment:
use time-constant comparison please
```suggestion
    head(:unauthorized) unless ActiveSupport::SecurityUtils.secure_compare(calculated_hmac, hmac_header)
```